File: app/src/main/java/com/route4me/trucksurvey/MainActivity.java
package com.route4me.trucksurvey;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.route4me.trucksurvey.model.TruckParams;
import com.route4me.trucksurvey.view.TruckSurveyView;

import java.util.Arrays;

import static com.route4me.trucksurvey.model.HazardousGood.Explosive;
import static com.route4me.trucksurvey.model.HazardousGood.Flammable;
import static com.route4me.trucksurvey.model.HazardousGood.Gas;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TruckSurveyView truckSurveyView = findViewById(R.id.truckSurveyView);
        truckSurveyView.bindData(TruckParams.newBuilder()
                .setTrailersCount(2)
                .setHeight(4.5f)
                .setWidth(5.6f)
                .setWeight(45.8f)
                .setWeightPerAxle(3.2f)
                .setMaxAllowedWeight(89.8f)
                .setIsDifficultTurnsAllowed(true)
                .setIsTunnelsAllowed(false)
                .setHazardousGoods(Arrays.asList(Explosive, Gas, Flammable))
                .build());
    }
}

File: app/src/main/java/com/route4me/trucksurvey/MainActivity.kt
package com.route4me.trucksurvey

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.Window
import androidx.appcompat.app.AppCompatActivity
import com.route4me.survey.TruckSurveyActivity
import com.route4me.survey.TruckSurveyActivity.SUBMIT_DATA_CODE
import com.route4me.survey.TruckSurveyActivity.TRUCK_PARAMS_KEY
import com.route4me.survey.model.TruckParams

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        supportActionBar!!.hide()
        setContentView(R.layout.activity_main)
        navigateToAddressesList()
    }

    private fun navigateToAddressesList() {
        startActivityForResult(Intent(this, TruckSurveyActivity::class.java), SUBMIT_DATA_CODE)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == SUBMIT_DATA_CODE) {
            val params = data!!.extras!!.getParcelable<TruckParams>(TRUCK_PARAMS_KEY)
            Log.d("TEST", "Parameters: $params")
        }
    }
}